src/gpio.rs

```diff
@@ -699,6 +695,19 @@ macro_rules! gpio {
 
                     #[cfg(feature = "unproven")]
                     impl<MODE> toggleable::Default for $PXi<Output<MODE>> {}
+
+                    impl<MODE> InputPin for $PXi<Output<MODE>> {
```

Comment:
Almost done, this also has to be implemented at line 231 for `PXx` which is a "fully reased pin", line 425 `$PXx`, which is a partially erase pin. :) The implementation is almost the same. The compiler will scream the difference to you, or you could look at the other implementations :wink:

Reply:
Minor nitpick: I think I would move this below 

```rust
                    impl<MODE> OutputPin for $PXi<Output<MODE>> {
```

at line 668, because this is another implementation for `Output`, so this position would make more sense.